module/platform/Webots/src/Webots.cpp

```diff
@@ -0,0 +1,257 @@
+
+        // Tell webots who we are
+        send_player_details(fd, global_config);
+
+        on<IO>(fd, IO::READ | IO::WRITE | IO::ERROR | IO::CLOSE).then([this, fd]() {
+            // Receiving
+
+            // Get the size of the message
+            uint32_t Nn;
+            if (recv(fd, &Nn, sizeof(Nn), 0 != sizeof(Nn))) {
+                log<NUClear::ERROR>("Failed to read message size from TCP connection");
+                return;
+            }
+
+            uint32_t Nh = ntohl(Nn);  // Convert from network endian to host endian
+
+            // Get the message
+            std::vector<char> data(Nh, 0);
+            if (uint64_t(recv(fd, data.data(), Nh, 0)) != Nh) {
+                log<NUClear::ERROR>("Failed to read message from TCP connection");
+                return;
+            }
+
+            // Deserialise the message into a neutron
+            SensorMeasurements msg = NUClear::util::serialise::Serialise<SensorMeasurements>::deserialise(data);
+            
+            translate_and_emit_sensor(msg);
+        });
+
+        on<Every<1, std::chrono::seconds>>().then([this, fd]() {
+            // Sending
+            std::vector<char> data = NUClear::util::serialise::Serialise<ActuatorRequests>::serialise(to_send);
+            uint32_t N             = htonl(data.size());
+            send(fd, &N, sizeof(N), 0);
+            send(fd, data.data(), N, 0);
+        });
+    });
+
+    // Create the message that we are going to send.
+    on<Trigger<ServoTargets>>().then([this](const ServoTargets& commands) {
+        // Maybe keep the `ServoTarget`s we have not sent yet, instead of just overriding them.
+        to_send.motor_positions.clear();
+        to_send.motor_velocities.clear();
+        to_send.motor_torques.clear();
+
+        // Store each ServoTarget to send in the next lot
+        for (auto& command : commands.targets) {
+            MotorPosition position_msg;
+            position_msg.name     = command.id;
+            position_msg.position = command.position;
+            to_send.motor_positions.push_back(position_msg);
+
+            // TODO(cmurtagh) work out if gain is velocity or force
+            MotorVelocity velocity_msg;
+            velocity_msg.name     = command.id;
+            velocity_msg.velocity = command.gain;
+            to_send.motor_velocities.push_back(velocity_msg);
+
+            MotorTorque torque_msg;
+            torque_msg.name   = command.id;
+            torque_msg.torque = command.torque;
+            to_send.motor_torques.push_back(torque_msg);
+
+            // MotorPID ? Do we need to send this?
+        }
+    });
+}
+
+void Webots::send_player_details(const int& fd, const GlobalConfig& player_details) {
+    // TODO(cameron) resend if failes
+    std::vector<char> data = NUClear::util::serialise::Serialise<GlobalConfig>::serialise(player_details);
+    uint32_t N             = htonl(data.size());
+    send(fd, &N, sizeof(N), 0);
+    send(fd, data.data(), N, 0);
```

Comment:
Should we keep this consistent?
```suggestion
    uint32_t Nn             = htonl(data.size());
    send(fd, &Nn, sizeof(Nn), 0);
    send(fd, data.data(), Nn, 0);
```